### ISSUE TITLE
What's the actor :name good for / Mistake in doc (?

### ISSUE THREAD

**Issue by monkeyjunglejuice** [mdbergmann/cl-gserver#98]
Hi Manfred, I'm working through Sento's documentation right now. I wonder what the :name of an actor is good for – I've been following the documentation <https://mdbergmann.github.io/cl-gserver/index.html>, and I stumbled upon the section where it says:
"Now we want to create actors."
``` common-lisp
  (actor-of *system* :name "answerer"
                   :receive
          (lambda (msg)
            (let ((output (format nil "Hello ~a" msg)))
              (reply output))))
```
Even though the doc says "Notice that the actor is not assigned to a variable (but you can)" The further text implies that `:name "answerer"` is creating a special variable `*answerer*`, which seems not to be the case actually:

"Now, when sending a message using 'ask' pattern to the above actor like so:"
``` common-lisp
(? *answerer* "FooBar")
```
That only works when I wrap the `(actor-of ...)` form into a `defparameter`. So I wonder if I can somehow send a message to an actor using the `:name` of that actor alone?

**Comment by mdbergmann** [comment]
You could use the name, but not directly, only using `find-actors`: https://mdbergmann.github.io/cl-gserver/index.html#SENTO.ACTOR-CONTEXT:FIND-ACTORS%20%28METHOD%20NIL%20%28SENTO.ACTOR-CONTEXT:ACTOR-CONTEXT%20T%29%29

In the case of the readme it would probably be easiest to just assign the actor to a special variable. I will have a look and fix it.

**Comment by monkeyjunglejuice** [comment]
Thanks :) So the `:name` doesn't correspond 1:1 to Erlang's PIDs, right? (sorry I don't know any Akka stuff – only Elixir, Erlang and Gleam as a starting point)

**Comment by mdbergmann** [comment]
Yes, `:name` just identifies an actor within an actor-context. actor-context is basically just an actor container. Actor names have to be unique within a context. An actor itself also represents a context. So it's possible to add 'child' actors whose name must be unique.
Globally (within an actor-system) actors are identified hierarchically by their name which builds up a path. The root context path is always "/user".

**Comment by mdbergmann** [comment]
I see you are looking also into LFE. I've developed a rudimentary LSP server for it some time ago (https://github.com/mdbergmann/lfe-ls).
That's also a great option IMO.

**Comment by monkeyjunglejuice** [comment]
> I see you are looking also into LFE. I've developed a rudimentary LSP server for it some time ago (https://github.com/mdbergmann/lfe-ls). That's also a great option IMO.

Yeah I realized that a while ago :) I just couldn't compile the LS; I've [opened an issue](https://github.com/mdbergmann/lfe-ls/issues/10#issue-2714103164) and attached the console output from my compilation attempts.

Besides that, LFE is pretty cool – at first I was quite thrown off by the "documentation"; but actually LFE is quite a small Lisp, and super close to Erlang's semantics anyway – hence the lack of documentation is not as much of an issue as I first thought it was.

I just wish it would have a bit more of a coordinated team behind it – Robert seems veeery busy (no surprise here), and it looks like he's the only one making *all the decisions*, no matter how small – which seems to slow things down.

**Comment by mdbergmann** [comment]
OK, thanks for the issue report, I'll check it out. Didn't work on it for a bit.

I think other than Robert there is only oubiwann (can't remember his real name right now) who helps with the tooling web page documentation and such. But it's basically a two man show.

Anyhow, LFE is of course closer to Erlang and Erlang actor model is quite different to Sento's. Actually, the first experiments and the first version was much closer to how Erlang actors work, from API point of view. The `ActorCell` in Sento still knows `call` and `cast`.
But since the Erlang runtime is much different to Common Lisps (or also Java's) it made only little sense to lean it on that.
So if you're coming from Erlang/OTP, then Sento is very different.

**Comment by mdbergmann** [comment]
Alright, I've changed the readme, please take a look.

**Comment by monkeyjunglejuice** [comment]
> Alright, I've changed the readme, please take a look.

I'll pick up the track through the Sento doc in 1–3 days again :) Right now I'm checking out your the LFE language server, and integrate it into my Emacs config.
